src/scenechange/mod.rs

```diff
@@ -152,33 +224,80 @@ impl SceneChangeDetector {
 
   /// Run a comparison between two frames to determine if they qualify for a scenecut.
   ///
-  /// The current algorithm detects fast cuts using changes in colour and intensity between frames.
-  /// Since the difference between frames is used, only fast cuts are detected
-  /// with this method. This is intended to change via https://github.com/xiph/rav1e/issues/794.
+  /// The standard algorithm uses block intra and inter costs
+  /// to determine which method would be more efficient
+  /// for coding this frame.
+  ///
+  /// The fast algorithm detects fast cuts using a raw difference
+  /// in pixel values between the frames.
+  /// It does not handle pans well, but the scene flash detection compensates for this
+  /// in many cases.
   fn has_scenecut<T: Pixel>(
-    &self, frame1: &Frame<T>, frame2: &Frame<T>,
-  ) -> bool {
-    let mut len = frame2.planes[0].cfg.width * frame2.planes[0].cfg.height;
-    let mut delta = 0;
-
-    delta += self.delta_in_planes(&frame1.planes[0], &frame2.planes[0]);
+    &self, frame1: Arc<Frame<T>>, frame2: Arc<Frame<T>>, frameno: u64,
+    keyframes: &BTreeSet<u64>,
```

Comment:
Would be better to not pass the set around, computing the last keyframe and passing it around usually works better (see my refactor)